### ISSUE TITLE
Assignment from yield reports a processing error

### ISSUE THREAD

**Issue by Nessworthy** [phpmd/phpmd#804]
- PHPMD version: 2.8.1 / 2.8.1snapshot202002241446
- PHP Version: 7.4.8 (both)
- Installation type: composer / phar
- Operating System / Distribution & Version: Windows 10

Also tested on `2.8.1` from source at `cf8677bbd8c291f7f6ed6437f5c941a49ebc2473`

### Current Behavior

Running phpmd reports the following processing error:

`Unexpected token: }, line: 6, col: 1, file: /path/to/script.php.`

### Expected Behavior

No processing errors.

### Steps To Reproduce:

Using the following script:

```php
<?php declare(strict_types=1);

function test_function(): Generator
{
    $result = yield true;
}
```

And running `phpmd script.php text cleancode`

or

`php phpmd.phar script.php text cleancode`

## Checks before submitting
* [X] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
    * Another prior issue around `yield`, from a previous version reporting and a different error, but it is closed off.
* [X] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [X] I have added every step to reproduce the bug.
* [X] If possible I added relevant code examples.
* [X] This issue is about 1 bug and nothing more.
* [X] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

**Comment by kylekatarnls** [comment]
Sure PHPMD should handle it, but as for my personal opinion, having an output (yield) and an assignation ($result =) in the same line of code is a code smell.

A clean code would be:

```php
<?php declare(strict_types=1);

function test_function(): Generator
{
    $result = true;

    yield $result;
}
```

**Comment by Nessworthy** [comment]
It's possible to need to make use of both and output and assignment, like

```php
<?php declare(strict_types=1);

function keep_incrementing_number(int $number) {
    $increment = 0;
    while (true) {
        $number += $increment;
        $increment = yield $number;
    }
}

$generator = keep_incrementing_number(0);

$generator->current(); // Returns 0
$generator->send(1);   // Returns 1
$generator->send(10);  // Returns 11
$generator->send(100); // Returns 111
```

Forgive the stupid example 😅

**Comment by kylekatarnls** [comment]
Indeed, I get the point.

**Comment by bug-tape** [comment]
Similar problem when using this short cut to stop a generator:
```php
<?php
function some_generator() {
    return yield "last";
  }
}
```

**Comment by wietsewarendorff** [comment]
Any progress on this?

**Comment by kylekatarnls** [comment]
Cause found, you can test the fix with:

```shell
composer require "pdepend/pdepend:dev-fix/yield-assignment as 2.16.0"
```
